plonky2/src/gates/base_sum.rs

```diff
@@ -192,6 +193,23 @@ impl<F: RichField, const B: usize> SimpleGenerator<F> for BaseSplitGenerator<B>
     }
 }
 
+/// Returns the largest `i` such that `base**i < n`.
```

Comment:
Yeah maybe call it `log_floor` and put it near `log2_ceil`?

(It seems like the current behavior is `ceil(log(n, base)) - 1` rather than `floor(log(n, base))`, but we could switch to the latter - seems a bit more intuitive and it should work for the caller above.)

Reply:
Sounds good, I'll move it to `util`. 
I think `log_floor` makes sense as it computes `floor(log(n-1, base))`.

Reply:
True but the `- 1` seems unexpected to me, wouldn't one expect `log_floor` to be `floor(log(n, base))`?

Reply:
Hm true, I changed it the the expected behavior.